src/mpyl/utilities/s3/__init__.py

```diff
@@ -15,17 +15,18 @@
 
 @dataclass
 class S3ClientConfig:
-    service_name = "s3"
-    region_name = "eu-central-1"
-    access_key_id: Optional[str]
-    secret_access_key: Optional[str]
     bucket_name: str
     bucket_root_path: str
+    service_name: str = field(default="s3")
+    region_name: str = field(default="eu-central-1")
+    access_key_id: Optional[str] = field(default=None)
+    secret_access_key: Optional[str] = field(default=None)
 
     def __init__(self, run_properties: RunProperties, bucket_name: str):
         s3_connection_info = run_properties.config.get("s3", None)
-        self.access_key_id = s3_connection_info.get("accessKeyId")
-        self.secret_access_key = s3_connection_info.get("secretAccessKey")
+        if s3_connection_info is not None:
+            self.access_key_id = s3_connection_info.get("accessKeyId", None)
```

Comment:
since you removed the 'required' check in the schema, shouldn't we now do a check here that they are filled?

Reply:
my idea is that there is no required check anymore. You can define them in the config or you can have any of the local setups for AWS credentials. If you don't define them and don't have the local auth setup then the client creation will throw and exception

Reply:
What do you mean with "you can have any of the local setups for AWS credentials"? I'm just thinking that if the client below is used without these credentials it's probably better to throw our own error beforehand already instead of depending on whatever unauthenticated error we get back

Reply:
I think it makes sense to rely on Boto3 since it automatically searches for credentials in a very [flexible way](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html).  For us to check in advance we'd have to do the same checks Boto3 already does which seems unnecessary if we can just let it do the checks.